internal/pkg/api/envoy/envoy-proxy.go

```diff
@@ -207,21 +210,16 @@ func (p *envoyExtAuthzGrpcServer) Check(ctx context.Context, req *ext_authz.Chec
 		httpRequest.Header.Set(headerKey, headerValue)
 	}
 
-	// Add unique identifier for logging purpose
-	httpRequest = utilInt.AssignRequestUID(httpRequest)
-	uid := utilInt.GetRequestUID(httpRequest)
-	logging.LogForComponentAndUID("envoyExtAuthzGrpcServer", uid).Infof("Received Envoy-Ext-Auth-Check to URL: %s", httpRequest.RequestURI)
-
 	w := telemetry.NewInMemResponseWriter()
 	(*p.compiler).ServeHTTP(w, httpRequest)
 
 	resp := &ext_authz.CheckResponse{}
 	switch w.StatusCode() {
 	case http.StatusOK:
-		logging.LogForComponentAndUID("envoyExtAuthzGrpcServer", uid).Infoln("Decision: ALLOW")
+		logging.LogAccessDecision(p.cfg.AccessDecisionLogLevel, path, r.GetMethod(), time.Since(startTime).String(), "ALLOW", "envoyProxy")
 		resp.Status = &rpc_status.Status{Code: int32(code.Code_OK)}
 	case http.StatusForbidden:
-		logging.LogForComponentAndUID("envoyExtAuthzGrpcServer", uid).Infoln("Decision: DENY")
+		logging.LogAccessDecision(p.cfg.AccessDecisionLogLevel, path, r.GetMethod(), time.Since(startTime).String(), "DENY", "envoyProxy")
```

Comment:
As far as I see it right now, the envoy-proxy and the policy-compiler are both logging the access decision. Therefore I thing removing the logs from the envoy-proxy is the best option to avoid duplications (the rest-proxy doesn't log the decisions either).

Removing the Envoy-Proxy from Kelon will also be an issue for the next year (as I discussed with Matthias right now) so you shouldn't put too much effort in this ;D.